qa-tests-backend/src/test/groovy/BaseSpecification.groovy

```diff
@@ -259,13 +259,19 @@ class BaseSpecification extends Specification {
         //Always make sure to revert back to the allAccessToken before each test
         resetAuth()
 
-        if (ClusterService.isEKS() && System.currentTimeSeconds() > orchestratorCreateTime + 600) {
+        if (ClusterService.isEKS()) {
             // Avoid EKS k8s client time out which occurs at approx. 15 minutes.
-            orchestrator = OrchestratorType.create(
-                    Env.mustGetOrchestratorType(),
-                    Constants.ORCHESTRATOR_NAMESPACE
-            )
-            orchestratorCreateTime = System.currentTimeSeconds()
+            synchronized(orchestrator) {
+                // synchronized() because orchestrator would be shared amongst
+                // concurrent feature threads.
+                if (System.currentTimeSeconds() > orchestratorCreateTime + 600) {
+                    orchestrator = OrchestratorType.create(
+                            Env.mustGetOrchestratorType(),
+                            Constants.ORCHESTRATOR_NAMESPACE
+                    )
+                    orchestratorCreateTime = System.currentTimeSeconds()
```

Comment:
I'm not sure if this is the correct solution. Maybe we should use https://spockframework.org/spock/docs/2.0-M4/parallel_execution.html#_resource_locks

Reply:
BTW, why do we need to recreate orchestrator every test?

Reply:
> BTW, why do we need to recreate orchestrator every test?

For EKS the k8s client times out after 900s. So with tests (+ failure debug gathering) a check per test feature for EKS is needed.

Reply:
> I'm not sure if this is the correct solution. Maybe we should use https://spockframework.org/spock/docs/2.0-M4/parallel_execution.html#_resource_locks

In this situation `synchronized(orchestrator)` works. `orchestrator` being a special spock @Shared means there is a separate instance per Spec. With parallel Specs there will be no lock and each will renew its orchestrator if needed. With parallel Features in a Spec only one Feature setup() will renew the client. 
Note: parallel test Features is probably not something we will try in the immediate future.